### ISSUE TITLE
util.serialize accepts a different format than ipfs object put

### ISSUE THREAD

**Issue by mitar** [ipld/js-ipld-dag-pb#27]
I can do:

```
echo '{"Data":"{\"Luck\":0.4,\"Proof\":\"<TEE signature>\"}","Links":[{"Hash":"QmUxD5gZfKzm8UN4WaguAMAZjw2TzZ2ZUmcqm2qXPtais7","Name":"payload","Size":819}]}' | ./ipfs object put
added QmYe6tCaRi5M2b64uTU4ikugBQvAq2jBrwBT1rqppvNFpi
```

But if I try to pass the same object to `util.serialize`, it does not really work, because `toProtoBuf` expects lower case properties.

**Comment by mitar** [comment]
Or it seems `ipfs object put` normalizes the input object. I can use also lowercase properties.

Maybe this package should normalize in the same way: accept lower or upper case properties, accept both `hash` and `multihash` for hashes, `size` and `tsize`.

**Comment by diasdavid** [comment]
> Maybe this package should normalize in the same way: accept lower or upper case properties, accept both hash and multihash for hashes, size and tsize.

Sounds good to me, would you like to submit a PR with it?

**Comment by mitar** [comment]
I think [based on your comment here](https://github.com/ipld/js-ipld-dag-pb/issues/28#issuecomment-287231216) that this might be the same issue. `util.serialize` does not normalize these field names because it expects a dag node instance anyway, not a plain object. So probably this [normalization code should be abstracted out](https://github.com/ipld/js-ipld-dag-pb/blob/master/src/dag-node/create.js#L41). And then used for both create and serialization.

**Comment by diasdavid** [comment]
that and also the normalization factor too.

**Comment by vmx** [comment]
This should be fixed with 0.14. If not, please re-open this issue.
